### ISSUE TITLE
Search options do not save when searching from Help

### ISSUE THREAD

**Issue by rohitab** [FarGroup/FarManager#612]
### Far Manager version

3.0.6094.0 x64

### OS version

10.0.22623.1180

### Steps to reproduce

1. Restart _Far_. This is required to reset search options.
2. Press `F1` to launch _Help_ from any Panel
3. Press `F7` to open _Search_ dialog
4. Enable the option _Case sensitive_
5. Press `Enter` to search
6. Press `Esc` to close _Help_
7. Press `F1` to relaunch _Help_
8. Press `F7` to open _Search_ dialog

### Expected behavior

_Case sensitive_ search option should be enabled.

### Actual behavior

_Case sensitive_ search option is disabled. Note that this happens with _Whole words_, _Regular expressions_ and _Fuzzy search_ options as well.

Next, please do the following:
1. Open the Viewer or Editor
2. Press `F7` to open the _Search_ dialog
3. Enable the _Case sensitive_ option
4. Press `Enter` to search
5. Ignore any error messages and close the Viewer or Editor
6. From any Panel press `F1` to launch _Help_
7. Press `F7` to open the _Search_ dialog
8. Note that _Case sensitive_ is now enabled
9. Turn off the _Case sensitive_ option
10. Press `Enter` to search
11. Press `Esc` to close _Help_
12. Press `F1` to relaunch _Help_
13. Press `F7` to open _Search_ dialog
14. Note that _Case sensitive_ is still enabled, even though it was turned off in Step 9 above.

To summarize, the search options in _Help_ are always initialized based on the last state from the Viewer/Editor. The changes made to the search options from _Help_ are not saved.

**Comment by MKadaner** [comment]
Thank you for reporting, @rohitab. Yes, this is how it works. The fix would be simple. As usual, I am very wary about changing the behavior which has been there for a long time.

Not saving search options, including search string, on closing Help may have been very intentional. Help and Editor / Viewer are two very different contexts. I can imagine a scenario when I am analyzing something juggling Find File and several Editors and keep (a part of) the context of my investigation in the search term available in all Search dialogs everywhere. If during such exercise I happen to open help and search for something, I will lose my context if Help overrides the last search string. Admittedly, this is less of an issue with search history available.

@alabuzhev, what do you think? Maybe we should keep a separate Help Search context between Help invocations? This way, something will be preserved, and it would be obvious to the user that it's intentionally _not_ the same as Editor / Viewer context.

**Comment by alabuzhev** [comment]
> Maybe we should keep a separate Help Search context between Help invocations?

Makes sense, Help has nothing to do with Viewer/Editor.

**Comment by MKadaner** [comment]
OK, please assign this one to me. I'll fix it after integrating unified search dialog into Viewer. Let's track _**saving / restoring Regex option**_ here as well. It's all will naturally happen in one go.

**Comment by MKadaner** [comment]
Fixed in 6135 (#678)

**Comment by rohitab** [comment]
Thanks! Verified in version `3.0.6136.0 x64`.
